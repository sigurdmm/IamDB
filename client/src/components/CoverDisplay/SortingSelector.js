import PropTypes from 'prop-types';
import React from 'react';

const SortingSelector = ({
  sortingMethods,
  onSort,
}) => {
  // If onSort is false don't render SortingSelector
  // || typeof onSortDirection === 'undefined'
  if (typeof onSort === 'undefined') {
    return null;
  }

  return <div>
    <select name="select" onChange={event => onSort(event.target.value)}>
    {sortingMethods.map((method, i) => (
      <option
        key={i}
        value={method.value}
      >
        {method.label}
      </option>))}
    </select>
  </div>;
};

SortingSelector.propTypes = {
  onSort: PropTypes.func,
  sortingMethods: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
};

export default SortingSelector;
