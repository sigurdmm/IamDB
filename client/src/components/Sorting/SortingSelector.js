import PropTypes from 'prop-types';
import React from 'react';

const SortingSelector = ({
  sortingMethods,
  onSort,
}) => <>
    <select
      name="select"
      className='sortingbar__type'
      onChange={event => onSort(event.target.value)}
    >
    {sortingMethods.map((method, i) => (
      <option
        key={i}
        value={method.value}
      >
        {method.label}
      </option>))}
    </select>
  </>;

SortingSelector.propTypes = {
  onSort: PropTypes.func.isRequired,
  sortingMethods: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
};

export default SortingSelector;
