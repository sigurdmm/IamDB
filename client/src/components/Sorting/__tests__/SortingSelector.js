import { shallow } from 'enzyme';

import SortingSelector from '../SortingSelector';


describe('<SortingSelector/>', () => {
  it('should render properly', () => {
    const wrapper = shallow(<SortingSelector
      onSort={jest.fn()}
      sortingMethods={ [{ label: 'label', value: 'value' }] }
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
