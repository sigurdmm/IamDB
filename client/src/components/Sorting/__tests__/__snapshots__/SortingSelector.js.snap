// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortingSelector/> should render properly 1`] = `
<Fragment>
  <select
    className="sortingbar__type"
    name="select"
    onChange={[Function]}
  >
    <option
      key="0"
      value="value"
    >
      label
    </option>
  </select>
</Fragment>
`;
